config/stp.py

```diff
@@ -0,0 +1,951 @@
+
+
+def is_valid_max_age(ctx, max_age):
+    if max_age not in range(STP_MIN_MAX_AGE, STP_MAX_MAX_AGE + 1):
+        ctx.fail("STP max age value must be in range 6-40")
+
+
+def is_valid_bridge_priority(ctx, priority):
+    if priority not in range(STP_MIN_BRIDGE_PRIORITY, STP_MAX_BRIDGE_PRIORITY + 1):
+        ctx.fail("STP bridge priority must be in range 0-61440")
+    if priority % 4096 != 0:
+        ctx.fail("STP bridge priority must be multiple of 4096")
+
+
+def validate_params(forward_delay, max_age, hello_time):
+    if (2 * (int(forward_delay) - 1)) >= int(max_age) >= (2 * (int(hello_time) + 1)):
+        return True
+    else:
+        return False
+
+
+def is_valid_stp_vlan_parameters(ctx, db, vlan_name, param_type, new_value):
+    stp_vlan_entry = db.get_entry('STP_VLAN', vlan_name)
+    cfg_vlan_forward_delay = stp_vlan_entry.get("forward_delay")
+    cfg_vlan_max_age = stp_vlan_entry.get("max_age")
+    cfg_vlan_hello_time = stp_vlan_entry.get("hello_time")
+    ret_val = False
+    if param_type == parameter_forward_delay:
+        ret_val = validate_params(new_value, cfg_vlan_max_age, cfg_vlan_hello_time)
+    elif param_type == parameter_max_age:
+        ret_val = validate_params(cfg_vlan_forward_delay, new_value, cfg_vlan_hello_time)
+    elif param_type == parameter_hello_time:
+        ret_val = validate_params(cfg_vlan_forward_delay, cfg_vlan_max_age, new_value)
+
+    if ret_val is not True:
+        ctx.fail("2*(forward_delay-1) >= max_age >= 2*(hello_time +1 ) not met for VLAN")
+
+
+def is_valid_stp_global_parameters(ctx, db, param_type, new_value):
+    stp_global_entry = db.get_entry('STP', "GLOBAL")
+    cfg_forward_delay = stp_global_entry.get("forward_delay")
+    cfg_max_age = stp_global_entry.get("max_age")
+    cfg_hello_time = stp_global_entry.get("hello_time")
+    ret_val = False
+    if param_type == parameter_forward_delay:
+        ret_val = validate_params(new_value, cfg_max_age, cfg_hello_time)
+    elif param_type == parameter_max_age:
+        ret_val = validate_params(cfg_forward_delay, new_value, cfg_hello_time)
+    elif param_type == parameter_hello_time:
+        ret_val = validate_params(cfg_forward_delay, cfg_max_age, new_value)
+
+    if ret_val is not True:
+        ctx.fail("2*(forward_delay-1) >= max_age >= 2*(hello_time +1 ) not met")
+
+
+parameter_forward_delay = 1
+parameter_hello_time = 2
+parameter_max_age = 3
+parameter_bridge_priority = 4
+
+
+def get_max_stp_instances():
+    return PVST_MAX_INSTANCES
+    # below part is not yet required for new updates
+    # state_db = SonicV2Connector(host='127.0.0.1')
+    # state_db.connect(state_db.STATE_DB, False)
+    # max_inst = state_db.get(state_db.STATE_DB, "STP_TABLE|GLOBAL", "max_stp_inst")
+    # if max_inst == "":
+    #    return PVST_MAX_INSTANCES
+    # if max_inst != None and max_inst != 0 and max_inst < PVST_MAX_INSTANCES:
+    #    return max_inst
+    # else:
+    #    return PVST_MAX_INSTANCES
```

Comment:
remove unused part. Also, if this is just returning a constant, can we just reuse the constant.

Reply:
Addressed